babel.config.js

```diff
@@ -0,0 +1,64 @@
+// @noflow
+
+/**
+ * The configuration for babel transpilation.
+ *
+ * Here we ensure that babel retranspiles files if the node version or the
+ * node environment are different.
+ */
+
+/**
+ * This file will not get transpiled, so we have to be cautious in the syntax
+ * we use.
+ */
+// eslint-disable-next-line import/no-commonjs
+module.exports = function(api) {
+    /**
+     * Determine the major version of NodeJS that is executing.
+     * The version is of the form v8.16.2, for example.
+     */
+    const nodeMajorVersion = process.version.split(".")[0].replace("v", "");
+    // eslint-disable-next-line no-console
+    console.log(`Transpiling for NodeJS ${nodeMajorVersion}`);
+
+    /**
+     * Cache based on the major node version and the environment.
+     */
+    api.cache.using(() => `${nodeMajorVersion}:${process.env.NODE_ENV}`);
```

Comment:
Huh! Not familiar with this API, very interesting! I wonder if we should be using this in webapp?

Reply:
Yeah, I was looking into making it switch based on Node versions so I could properly investigate things and they add this `api.cache` thing. I learned something new :)